packages/components/src/ui/context/context-connect.ts

```diff
@@ -39,19 +49,19 @@ export function contextConnect< P >(
 	namespace: string,
 	options: ContextConnectOptions = {}
 ): WordPressComponentFromProps< P > {
-	const { memo: memoProp = false } = options;
+	const { memo: memoProp = false, noForwardRef } = options;
+
+	let WrappedComponent = noForwardRef ? Component : forwardRef( Component );
```

Comment:
It would be cool if we could automatically detect the presence of a ref argument without having to manually set the `noForwardRef` option. However, I think it's impossible because we cannot reliably detect **in JS** whether a component function takes one argument (props) or two (props & ref). If we can't detect in JS, we can't do this conditional.

The next best thing would be to detect **in TS** whether the `noForwardRef` boolean is **wrong**, given the signature of the component function. I tried a bunch of ways with generics and conditional types, but unfortunately could not get it to work! It's a nice-to-have if anyone wants to try figuring it out.

Reply:
Yay, I found out a better way — TypeScript will now throw an error if the ref stuff is wrong. I updated the PR description.